[g-tejas] iP

g-tejas opened the pull request:
# Miss Minutes Chatbot ⏰

> _"The best thing since sliced bread"_ - Mahatma Gandhi ([source](https://www.youtube.com/watch?v=dQw4w9WgXcQ))

## Why should you use this over other task managers?
- _Please use meeeee_ 👉👈 🥺
- Won't yell at you, very kind and sweet
- **Miss Minutes will try her very best**

## Roadmap
Here are some of the cutting edge new features I am currently working on
- [ ] Backend powered by AGI
- [ ] Collaboration with DeepMind and OpenAI
- [ ] Integration with Elon Musk's Neuralink chip (eliminate the need for UI entirely)

## How to run this?
```java
class MissMinutes {
    public static void main(String[] args) {
        System.out.println("I love 2103!");
    }
}
```

## How can I help?
I am glad you asked!

1. Fork the repository and clone it with `git clone <url of your fork>`
2. Make changes to your fork
3. Submit PR and please follow the conventions



~~hehe~~

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

g-tejas committed:
Renamed the chatbot from Duke to Miss Minutes

g-tejas committed:
Implemented a simple skeletal version that simply greets the user and exits

g-tejas committed:
Added echo functionality and a few helper methods for MissMinutes chatbot

g-tejas committed:
Added the ability to store tasks entered by the user and display them when requested with "list"

g-tejas committed:
Added a new Task class to replace String primitive, has status & description fields and respective helper methods

g-tejas committed:
Replaced String with Task, added ability to mark tasks as done and undone, and displaying their current status

g-tejas committed:
Fixed a typo in the reply message

g-tejas committed:
Added subclass Deadline with factory method

g-tejas committed:
Added subclass Event with factory constructor method

g-tejas committed:
Added subclass Todo with factory constructor method, no extra fields

g-tejas committed:
Added support for tracking three types of tasks, refactored code, added factory constructor for Task subclasses

g-tejas committed:
Complete A-TextUiTesting level

g-tejas committed:
Added MissMinutesException to deal with incorrect inputs entered by the user

g-tejas committed:
Added support for deleting tasks from list

g-tejas committed:
Used Java enums and added a string to enum parser, and adjusted the error handling

g-tejas committed:
Implement Serializable for classes

g-tejas committed:
Implement Level-7 to save tasks to disk

g-tejas committed:
Update Deadline, Event, to have DateTime

g-tejas committed:
Update EXPECTED.txt with DateTime changes

g-tejas committed:
Update test script to remove `tasks.bin`

g-tejas committed:
Complete Ui portion of A-MoreOOP

g-tejas committed:
Complete Storage portion of A-MoreOOP

g-tejas committed:
Completed TaskList portion of A-MoreOOP

g-tejas committed:
Complete Parser portion of A-MoreOOP

g-tejas committed:
Move classes to MissMinutes package

g-tejas committed:
Merge branch 'add-gradle-support'
Merge add-gradle-support into master

g-tejas committed:
Tweak build.gradle file to support MissMinutes

g-tejas committed:
Implement JUnit test for Deadline and Parser classes

g-tejas committed:
Update Jar file name

g-tejas committed:
Add JavaDoc comments to all classes and methods

g-tejas committed:
Refactor code to follow coding standard

g-tejas committed:
Add find command to search for tasks with keyword

g-tejas committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/MissMinutes/TaskList.java

g-tejas committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/MissMinutes/MissMinutes.java
#	src/main/java/MissMinutes/Parser.java

g-tejas committed:
Resolve merge conflict issue with changed directories and added more JavaDoc comments

g-tejas committed:
Increase JavaDoc coverage

hiivan commented on src/main/java/missminutes/MissMinutes.java:
```diff
@@ -0,0 +1,95 @@
+                    tasks.findTask(request);
+                    break;
+                case UNKNOWN:
+                        throw new MissMinutesException("Oh, I'm sowwy, I didn't undewstand dat. (>_<) Can I hewp wif sumthin' else, pwease? UwU");
```
Is this appropriate?

hiivan reviewed (commented):
uwu

g-tejas committed:
Complete A-Checkstyle increment

g-tejas committed:
Fix checkstyle violations

g-tejas committed:
Fix checkstyle violations and comments

g-tejas committed:
Add javafx functionality

g-tejas committed:
Add Varargs functionality

g-tejas committed:
Merge branch 'branch-A-Varargs'

xuelinglow commented on src/main/java/missminutes/Todo.java:
```diff
@@ -0,0 +1,41 @@
+     * @return The Todo object deserialized from the string
+     * @throws MissMinutesException If the given string is in incorrect format
+     */
+    public static Todo fromStr(String input) throws MissMinutesException {
```
Maybe can change the method name to something more specific e.g createTodoFromStr
Similar for the other types of task (deadline, event)

xuelinglow reviewed (commented):
Generally well done, I don't see much problems in terms of coding standards.  Good job!

g-tejas committed:
Change shadowJar source file

g-tejas committed:
Add assertions in key points of code

Added assertions to document important assumptions that should hold at
various points in the code.

g-tejas committed:
Improve code quality of Storage class

Cleaned up MissMinutes class by moving exception handling to
the Storage class. Improves readability of the MissMinutes class.

g-tejas committed:
Add stream functionality

Implemented the find command using streams to improve code
readability and performance.

g-tejas committed:
Merge pull request #1 from g-tejas/branch-A-Assertions

Complete A-Assertions

g-tejas committed:
Merge branch 'master' into branch-A-CodeQuality

g-tejas committed:
Merge pull request #2 from g-tejas/branch-A-CodeQuality

Complete A-CodeQuality

g-tejas committed:
Merge branch 'master' into branch-A-Streams

g-tejas committed:
Merge pull request #3 from g-tejas/branch-A-Streams

Complete A-Streams

g-tejas committed:
Implement C-BetterSearch

g-tejas committed:
Merge branch 'branch-BCD-Extension'

g-tejas committed:
Complete A-CI

g-tejas committed:
Add window title

g-tejas committed:
Add representative screenshot

g-tejas committed:
Update user guide

g-tejas committed:
Fix user guide
